src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -549,7 +549,7 @@ internal Manifest MergeManifests(List<Manifest> parsedManifests)
             }
 
             // Error out for any duplicated packages based on the top level properties of the package.
-            var distinctPackages = manifest.Packages.Distinct();
+            var distinctPackages = manifest.Packages.Distinct(new PackageIdComparer());
```

Comment:
We could also use `DistinctBy(package => package.Id, StringComparison.OrdinalIgnoreCase)`